### PR TITLE
Counter component

### PR DESCRIPTION
**Pull request by driesd**
schermafdruk 2017-09-19 10 30 36

**Out of scope:** Tooltip component

### CODE REVIEW

File: components/counter/theme.css [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+.medium {
+  min-height: 9px;
+  min-width: 9px;
+}
+
+.small {
+  min-height: 6px;
+  min-width: 6px;
```
Comment (nickwaelkens) [inline components/counter/theme.css]: **PP** Can we use css4 variables here instead? 
Example here: https://github.com/teamleadercrm/teamleader-ui/blob/button-v2/components/button/theme.css

File: components/counter/Counter.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      children,
+      className,
+      color,
+      size,
+      ...others
+    } = this.props;
+
+    const classes = cx(
+      theme.counter,
+      theme.monospaced,
```
Comment (lowiebenoot) [inline components/counter/Counter.js]: Seems like a leftover? The class isn't doing anything as far as I can see (although it actually has to be monospaced).

File: components/counter/Counter.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      theme[color],
+      theme[size],
+      {
+        [theme.circular]: !children,
+        [theme.rounded]: children,
```
Comment (lowiebenoot) [inline components/counter/Counter.js]: I would just make the css rounded by default, and only add the `circular` "variant" when necessary. This way you can drop the rounded class, and you end up with just circular being a variant.

File: components/counter/Counter.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      size,
+      ...others
+    } = this.props;
+
+    const classes = cx(
+      theme.counter,
+      theme[color],
+      theme[size],
+      {
+        [theme.rounded]: count,
```
Comment (lowiebenoot) [inline components/counter/Counter.js]: If the count is `0`, it will not show the rounded variant.

**Comment by lowiebenoot** [comment]
I'm also having some doubts about the color prop. 
Is the color functional as in e.g. active/not-active? Or is it really just as a color, and we're limiting it to 2 colors? If it's the 2nd one, we can keep it as a color prop, if it has some functionality, I would change it so active/non-active (or a better suitable name).

**Comment by stephen-tl** [comment]
I see this as functional color: active/Inactive. One thing that I will add to the spec in zeplin is a negative version of the inactive state. This can then be used in things like the sidenav. you can expect an update in 15 minutes

**Comment by stephen-tl** [comment]
After discussing this, we decided to make it possible to use multiple colors as a background. This to ensure flexibility later on. I'll update the spec later today.

File: components/counter/Counter.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      theme[color],
+      theme[size],
+      {
+        [theme.dark]: dark,
+        [theme.rounded]: typeof (count) !== 'undefined',
+      },
+      className,
+    );
+
+    const rest = omit(others, [
```
Comment (lowiebenoot) [inline components/counter/Counter.js]: `dark` is not in `others`, so you can ditch the `omit`.

**Pull request merged**